### DISCUSSION TITLE
YAML indent number of spaces

### DISCUSSION THREAD

**Discussion by macMikey** [trevordevore/levure#178] [Q&A]
I was trying to debug a yaml file I was creating for use with [yamlFileToArray](https://github.com/trevordevore/levure/wiki/Levure-API#yamlFileToArray). I noticed that indenting one space on child elements didn't work. So I went into the framework, but I ran into a bit of confusion.
Am I reading this correctly?
In the current dev commit (b96a8cc540db874d0d8d54b4fedc19e5d84aa81c), in **levureLoadAppConfig**, I see, on line 3468,
```
            put "    " before tLine
```
Which is supposed to "add back any stripped literal indents"
That is 4 spaces.

However, in **__PathElement**, line 3746, we have
```
    if xLine begins with "  " then
```
to determine if a line is indented a level. That is 2 spaces.

I thought that any level could be indented any number of spaces, provided that it was more spaces than the parent, and as long as all brothers and sisters were indented the same number of spaces.

Indenting a child two spaces per level does cause the array to be populated.
If we are going to use a set number of spaces to specify a child we should update the docs.

**Comment by trevordevore** [comment]
@macMikey Looks like the docs need updating. The parser is looking for 2 space indents. I'm not sure about the four spaces part. on line 3746. I would think that to properly support whatever indenting level is standard in the file that code to detect the indent level, store it, and then use it to check for/reinsert indents.

**Reply by macMikey** [reply]
I do not understand that comment.

**Reply by trevordevore** [reply]
@macMikey you said:

> If we are going to use a set number of spaces to specify a child we should update the docs.

The way the code is written requires 2 spaces for each level of indent. So docs need updating.

For the code to work the way you wanted it to work a code update would be required.

**Reply by macMikey** [reply]
So are you saying you are open to either option? Both modifying the code and modifying the docs are trivial. There is not another LC YAML library that I am aware of, and there has not been any other discussion of this topic, yet.
The four-space line is one that I do not understand. I do not know if that was intentional or a mistake.

**Comment by trevordevore** [comment]
I’m fine with a PR for either change. If you modify the code then I would want an example in the PR that shows the bug as well as how the update fixes it. I think @montegoulding wrote the YAML code originally. I can’t remember if I made any changes or not.
